GVFS/GVFS.Virtualization/Projection/Readme.md

```diff
@@ -0,0 +1,181 @@
+﻿# GitIndexProjection
+
+## Overview
+
+This document is to help give developers a better understanding of the `GitIndexProjection` class and associated classes and the design and architectural decisions that went into it. In simplest terms the purpose of the `GitIndexProjection` class is to parse the `.git/index` file and build an in memory tree representation of the directories and files that are used when a file system request comes from the virtual file system.  This helps VFSForGit be able to quickly return the results for file system to the user. VFSForGit uses the [skip worktree bit](https://git-scm.com/docs/git-update-index#_skip_worktree_bit) to know what to include in the projection data and what files git will be keeping up to date.  Currently VFSForGit only supports using [version 4 of the index](https://git-scm.com/docs/git-update-index#Documentation/git-update-index.txt---index-versionltngt).  Details on the index format and version 4 can be found [here](https://github.com/microsoft/git/blob/031fd4b93b8182761948aa348565118955f48307/Documentation/technical/index-format.txt).

```

Comment:
nit:
>  This helps VFSForGit be able to quickly return the results for file system to the user

Feels like we're missing a word in there somewhere. Maybe "This helps VFSForGit quickly return filesystem operations such as enumeration or on-demand hydration"?